File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        nums_xor=nums[0]
        res = 0^len(nums) 
        for i in range(1,len(nums)): 
            nums_xor=nums_xor ^ nums[i]
            res = res ^ i

        return nums_xor ^ res


        

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=nums.size();
        int xord=0;
        for(int i=0;i<=n;i++) xord^=i;
        for(auto& i:nums) xord^=i;
        return xord;
    }
};